### DISCUSSION TITLE
G4 doorbell gives rectangular picture instead of square

### DISCUSSION THREAD

**Discussion by jeroenvandisseldorp** [briis/unifiprotect#234] [Q&A]
Hi,

I'm a grateful user of this integration. Works terrific from the start.

Question on the camera picture: I'm using two cameras, one G3 dome and a G4 doorbell. Both live streams give me a rectangular view, but the G4 should give me a square picture instead. I've tried to get there by using the aspect_ratio option, but that only enlarges the frame with empty content, the picture of the doorbell stays squashed vertically.

Is there any way to get the square picture in a picture-entity?

Thanks
  Jeroen

**Comment by briis** [comment]
If you go in to the Unifi Protect Web App, and click on your doorbell camera. A Menu will pop-up to the right, and there you click on *Manage*. Then click on RTSP. For each stream that is activated you can see the image size: High: 1600x1200, Medium: 960x720 and Low: 480x360.

This integration will always select the highest image quality, so if High is enabled you will get an image of 1600x1200 and that will always be a rectangular view.

All the Cameras I have attached, will give a rectangular view due to the aspect ratio, and I don't believe you can do anything about that in Lovelace. My Doorbell looks like this in a Picture Glance card and I am happy with that.

<img width="971" alt="Skærmbillede 2021-05-18 kl  14 28 03" src="https://user-images.githubusercontent.com/24776272/118651037-62540780-b7e5-11eb-9199-a7be90d24b9c.png">

**Reply by jeroenvandisseldorp** [reply]
Thanks for your reply. Using your answer I checked the resolution, and indeed, the doorbell pictures seem more rectangular than they are in reality. So I checked with other cameras and found there is a difference still.

The aspect ratio of the doorbell is 1600x1200 = 1.33 aspect ratio. With other cameras this lies around 1.77, so significantly different from the doorbell's aspect ratio.

Both pictures get the same aspect ratio in HA however, being 1.77. I started noticing this because when people ring the doorbell, their faces seem distorted (compressed vertically) and I sometimes have difficulty recognising close friends on my phone with a relatively small picture area.

So rephrasing my question: is there a way to display the original 1.33 aspect ratio for doorbell's images?

**Comment by briis** [comment]
I might not be the right person to ask, as my integration only delivers the raw image. 
I am not sure when you see distorted images, is that in Lovelace or do you send yourself an image on your phone?

**Reply by jeroenvandisseldorp** [reply]
Both actually. It seems as if the raw image already has the distorted
aspect ratio. When I measure both my cameras in Lovelace (using taking
screenshot) I measure an aspect ratio of 1.77 with each one. The doorbell
snapshots I send to my phone when the doorbell is pressed also have the
same ratio. Even the picture of your garden has the same 1.77 ratio (you
can check by taking a screenshot and measuring x and y sizes). But when I
use the Unify Protect app on Android and take a screenshot there, then the
aspect ratio of the doorbell view is 1.33. In the live view there is
already a clear distinction in size between "normal cameras" and the
doorbell camera.

So the question becomes: are you getting a raw picture or stream for which
Protect determines the x and y sizes, or is your integration querying
Protect to get an image with a certain (assumed) aspect ratio?

**Comment by briis** [comment]
Hi Jeroen,
Well spotted! I just checked the code, and I actually set the Aspect Ratio when doing the snapshot with some predefined ratios. That code is from before the Doorbell came out, so it has not been a problem before. I will try to find a nice way of fixing this. Thanks.

EDIT: Could you please open an Issue, so that I can I am sure it gets fixed. Thanks

**Reply by jeroenvandisseldorp** [reply]
Great to hear you've found the cause! And thanks in advance for fixing. I opened an issue here: https://github.com/briis/unifiprotect/issues/235
